### ISSUE TITLE
error: this ‘else’ clause does not guard.

### ISSUE THREAD

**Issue by PikachuHy** [intel/caffe#202]
more info
```
/home/pikachu/swcontest/intel_caffe/src/caffe/layers/mkldnn_inner_product_layer.cpp: In member function ‘void caffe::MKLDNNInnerProductLayer<Dtype>::InitInnerProductBwd(const std::vector<caffe::Blob<Dtype>*>&, const std::vector<bool>&, const std::vector<caffe::Blob<Dtype>*>&)’:
/home/pikachu/swcontest/intel_caffe/src/caffe/layers/mkldnn_inner_product_layer.cpp:361:2: error: this ‘else’ clause does not guard... [-Werror=misleading-indentation]
  else
  ^~~~
/home/pikachu/swcontest/intel_caffe/src/caffe/layers/mkldnn_inner_product_layer.cpp:365:5: note: ...this statement, but the latter is misleadingly indented as if it is guarded by the ‘else’
     ipBwdData_desc.reset(new inner_product_backward_data::desc(init_bottom_md, init_weights_md, init_top_md));
     ^~~~~~~~~~~~~~
[ 40%] Building CXX object src/caffe/CMakeFiles/caffe.dir/layers/mkldnn_relu_layer.cpp.o
[ 40%] Building CXX object src/caffe/CMakeFiles/caffe.dir/layers/mkldnn_split_layer.cpp.o
cc1plus: all warnings being treated as errors
src/caffe/CMakeFiles/caffe.dir/build.make:1742: recipe for target 'src/caffe/CMakeFiles/caffe.dir/layers/mkldnn_inner_product_layer.cpp.o' failed
make[2]: *** [src/caffe/CMakeFiles/caffe.dir/layers/mkldnn_inner_product_layer.cpp.o] Error 1
make[2]: *** 正在等待未完成的任务....
CMakeFiles/Makefile2:335: recipe for target 'src/caffe/CMakeFiles/caffe.dir/all' failed
make[1]: *** [src/caffe/CMakeFiles/caffe.dir/all] Error 2
Makefile:129: recipe for target 'all' failed
make: *** [all] Error 2
```

**Comment by hshen14** [comment]
May I know which compiler did you use? I cannot reproduce the issue although the patch is fine.

**Comment by hufangjian** [comment]
https://github.com/PikachuHy/caffe/commit/756816db583fc8562f882b3f22792427e2b59010

it woks.

**Comment by reazulhoque** [comment]
I am getting the same error with gcc 7.2.1

**Comment by peteruhrig** [comment]
Same error with gcc 7.3.0 on Ubuntu 18.04

**Comment by hshen14** [comment]
Fixed it internally and will push to Github soon.

**Comment by cgaspart** [comment]
@hshen14 Soon ?

**Comment by hshen14** [comment]
Yes.

**Comment by juliebuquet** [comment]
Hello I have the same problem. Any News ?

**Comment by uhofemei** [comment]
How soon is soon? Caffe doesn't compile on Ubuntu 18.04 unless I turn the error off ;-)

**Comment by psyhtest** [comment]
Hi everyone,

You can now build Intel-Caffe from master with GCC using the following simple instructions from [Intel's ReQuEST artifact](https://github.com/ctuning/ck-request-asplos18-caffe-intel):

### Install Collective Knowledge
```
$ git clone http://github.com/ctuning/ck
$ export PATH=$PWD/ck/bin:$PATH
$ export PYTHONPATH=$PWD/ck:$PYTHONPATH
```
or simply:
```
$ sudo pip install ck
```

### Install Intel Caffe from master
```
$ ck pull repo:ck-request-asplos18-caffe-intel
$ ck install package:lib-caffe-intel-master-cpu
```
Tested with 3 GCC versions:
```
$ ck show env --tags=lib,intel,caffe
Env UID:         Target OS: Bits: Name:                        Version:       Tags:

2d58784a07aa1192   linux-64    64 BVLC Caffe framework (intel) master-a3d5b02 64bits,bvlc,caffe,compiled-by-gcc,compiled-by-gcc-8.1.0,host-os-linux-64,intel,lib,target-os-linux-64,v0,v0.0,vcpu,vmaster
18bd3430d346fabe   linux-64    64 BVLC Caffe framework (intel) master-a3d5b02 64bits,bvlc,caffe,compiled-by-gcc,compiled-by-gcc-6.4.0,host-os-linux-64,intel,lib,target-os-linux-64,v0,v0.0,vcpu,vmaster
5b9202a1f26b4107   linux-64    64 BVLC Caffe framework (intel) master-a3d5b02 64bits,bvlc,caffe,compiled-by-gcc,compiled-by-gcc-7.3.0,host-os-linux-64,intel,lib,target-os-linux-64,v0,v0.0,vcpu,vmaster
```

Hope this helps! Shout if you have any problems.

**Comment by Albert1990** [comment]
Hey Guys regarding this issue it has been fixed by turning the flag USE_MKL to be off while generating the files using cmake like the following:
first i installed python cafe:
pip3 install ck
then i reconfigured cmake like the following:
`cd <to your openpose dir>
mkdir build
cd build
cmake .. -DGPU_MODE="CPU_ONLY" -DDOWNLOAD_BODY_COCO_MODEL=ON -DDOWNLOAD_BODY_MPI_MODEL=ON -DPROFILER_ENABLED=ON -DWITH_OPENCV_WITH_OPENCL=ON -DWITH_OPENCV_WITH_OPENCL=ON -DUSE_MKL=OFF
`

**Comment by ftian1** [comment]
@Albert1990 your proposal is not related with the issue reported by poster.

@all, this in fact has been solved in 1.1.2 release. I would close this issue
